### PR TITLE
Add `token-stream-stress` benchmark.

### PR DESCRIPTION
**Pull request by nnethercote**
This will prevent the fix in https://github.com/rust-lang/rust/pull/65198 from regressing.

The benchmark is quite short-running, taking roughly 3x the time of `helloworld`. That's more than enough to detect regressions; prior to the fix it would have taken more than 100x longer.

@dtolnay, @alexcrichton: this benchmark is based on [this test case](https://github.com/alexcrichton/proc-macro2/tree/12bac84dd8d090d2987a57b747c7ae7bbeb8a3d0/benches/bench-libproc-macro). I hope that is ok. Let me know if you have ideas on how to improve this.

### CODE REVIEW

File: collector/benchmarks/token-stream-stress/perf-config.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "cargo_opts": "--bin token-stream-stress",
+    "runs": 1
```
Comment (Mark-Simulacrum) [inline collector/benchmarks/token-stream-stress/perf-config.json]: Is there a reason for running this just once? Speed? Generally I think we should try to keep the default of 3 runs and only adjust lower if it's a super long benchmark (which this doesn't seem to be).

Comment (nnethercote) [reply collector/benchmarks/token-stream-stress/perf-config.json]: No good reason. I think I unwittingly copied that `perf-config.json` file from one of the benchmarks that only runs once. I will change it on Monday.

**Comment by nnethercote** [comment]
@Mark-Simulacrum: I have removed the `runs=1` line from the `perf-config.json` file.

**Comment by Mark-Simulacrum** [comment]
Will merge once CI verifies this works (it will pass either way, but manual inspection of the log is possible).

**Comment by Mark-Simulacrum** [comment]
Uh, looks like CI actually failed -- probably a trailing comma in the perf config?

**Comment by nnethercote** [comment]
New version, with trailing comma removed.

**Pull request merged**